addons/stock/models/stock_rule.py

```diff
@@ -524,98 +511,82 @@ def _procure_orderpoint_confirm(self, use_new_cursor=False, company_id=None):
-                                continue
-                            if float_compare(op_product_virtual, orderpoint.product_min_qty, precision_rounding=orderpoint.product_uom.rounding) <= 0:
-                                qty = max(orderpoint.product_min_qty, orderpoint.product_max_qty) - op_product_virtual
-                                remainder = orderpoint.qty_multiple > 0 and qty % orderpoint.qty_multiple or 0.0
-
-                                if float_compare(remainder, 0.0, precision_rounding=orderpoint.product_uom.rounding) > 0:
-                                    qty += orderpoint.qty_multiple - remainder
-
-                                if float_compare(qty, 0.0, precision_rounding=orderpoint.product_uom.rounding) < 0:
-                                    continue
-
-                                qty -= substract_quantity[orderpoint.id]
-                                qty_rounded = float_round(qty, precision_rounding=orderpoint.product_uom.rounding)
-                                if qty_rounded > 0:
-                                    values = orderpoint._prepare_procurement_values(qty_rounded, **group['procurement_values'])
-                                    try:
-                                        with self._cr.savepoint():
-                                            #TODO: make it batch
-                                            self.env['procurement.group'].run([self.env['procurement.group'].Procurement(
-                                                orderpoint.product_id, qty_rounded, orderpoint.product_uom,
-                                                orderpoint.location_id, orderpoint.name, orderpoint.name,
-                                                orderpoint.company_id, values)])
-                                    except UserError as error:
-                                        self.env['stock.rule']._log_next_activity(orderpoint.product_id, error.name)
-                                    self._procurement_from_orderpoint_post_process([orderpoint.id])
-                                if use_new_cursor:
-                                    cr.commit()
-
-                        except OperationalError:
-                            if use_new_cursor:
-                                orderpoints_noprefetch += [orderpoint.id]
-                                cr.rollback()
-                                continue
-                            else:
-                                raise
+                orderpoint_context = orderpoint._procurement_from_orderpoint_get_context()
+                product_context = frozendict({**self.env.context, **orderpoint_context})
+                if product_context in orderpoints_contexts:
+                    orderpoints_contexts[product_context] |= orderpoint
+                else:
+                    orderpoints_contexts[product_context] = orderpoint
+
+            for orderpoint_context, orderpoints_by_context in orderpoints_contexts.items():
```

Comment:
orderpoint1 = product1 into WH/Stock/shelf1
orderpoint2 = product1 into WH/Stock
as you commit in batch, the two rr will create an procurement while only the orderpoint2 is enough right ? Don't know if this can be an issue or not